docs/in-depth-guide-scraper-functions.md

```diff
@@ -0,0 +1,319 @@
+'bbcgoodfood.com'
+```
+### `description() -> str`
+
+Returns a description of the recipe. This is normally a sentence or short paragraph describing the recipe. Often the website defines the description, but sometimes it has to be inferred from the page content.
+
+```py
+>>> scraper.description()
+'Let your little monsters do their worst, decorating these spooky Halloween treats'
+```
+
+### `image() -> str`
+
+Returns the URL to the main image associated with the recipe, usually a photograph of the completed recipe.
+
+```py
+>>> scraper.image()
+'https://images.immediate.co.uk/production/volatile/sites/30/2020/08/recipe-image-legacy-id-405483_12-cee017a.jpg?resize=768,574'
+```
+### `ingredients() -> List[str]`
+
+Returns the ingredients needed to make the recipe as a `list` of `str`. Each element of the list is usually a single sentence stating an ingredient, the required amount and any additional comments. The elements of the list should mirror the ingredients written on the website and should not include non-ingredient sentences such as sub-headings.
+
+```py
+>>> scraper.ingredients()
+[
+	'250g self-raising flour',
+	'25g cocoa powder',
+ 	'175g light muscovado sugar',
+ 	'85g unsalted butter , melted',
+ 	'5 tbsp vegetable or sunflower oil',
+ 	'150g pot fat-free natural yogurt',
+ 	'1 tsp vanilla extract',
+ 	'3 large eggs',
+ 	'85g unsalted butter , softened',
+ 	'1 tbsp milk',
+ 	'½ tsp vanilla extract',
+ 	'200g icing sugar , sifted',
+ 	'food colourings (optional)',
+ 	'sweets and sprinkles, to decorate'
+]
+```
+
+### `instructions() -> str`
+
+Returns a single `str` containing all instructions steps. Where a recipe has multiple instructions, each step is separated in the returned `str` by a newline character (`\n`).
+
+```py
+>>> scraper.instructions()
+'Heat oven to 190C/170C fan/gas 5 and line a 12-hole muffin tin with deep cake cases. Put all the cake ingredients into a large bowl and beat together with electric hand beaters until smooth. Spoon the mix into the cases, then bake for 20 mins until risen and a skewer inserted into the middle comes out dry. Cool completely on a rack. Can be made up to 3 days ahead and kept in an airtight container, or frozen for up to 1 month.\nFor the frosting, work the butter, milk and vanilla into the icing sugar until creamy and pale. Colour with food colouring, if using, then create your own gruesome monster faces using sweets and sprinkles.'
+```
+
+> [!IMPORTANT]
+>
+> Occasionally, a recipe will have steps that have new lines within them. At the moment, this library cannot distinguish between a newline within a step and a newline between steps.
+
+### `title() -> str`
+
+Returns the title of the recipes, usually a short sentence or phrase.
+
+```py
+>>> scraper.title()
+'Monster cupcakes'
+```
+
+### `total_time() -> int`
+
+Returns the total time required to complete the recipe, in minutes.
+
+```py
+>>> scraper.total_time()
+50
+```
+
+> [!NOTE]
+> Is the return type correct here?
```

Comment:
The `int` type does have the benefit of being straightforward to emit and consume -- and in some cases it's a direct typecast of the value that was presented on the HTML page.

On the other hand, I suppose [`timedelta`](https://docs.python.org/3/library/datetime.html#datetime.timedelta) could be a more structured alternative?

(I'm not sure I have strong opinions about this - maybe I should have, but I don't.  the only strong opinion I have is that if we do change it, we should migrate all three `*_time` fields during the `v15` migration)

Reply:
I wasn't sure if a recipe would ever specify a time to sub-minute resolution, in which case we would want to use `float`. 
`int` is probably fine, so I'll leave it like this.

There might be a job to do to check how well the existing scrapers we have align with the return types I've documented here.